File: src/main/javascript/crypto/src/libs/PersonalSignature.ts
import {AbstractSignature} from "./AbstractSignature";
import {KeyPair} from "./KeyPair";
import {SignatureUtility} from "./SignatureUtility";
import {uint8toString} from "./utils";

export class PersonalSignature extends AbstractSignature {
    static TYPE_OF_SIGNATURE = "personal";

    public fromMessage(keys: KeyPair, unprocessedMsg: Uint8Array) {
        super.keyMessageType(keys, unprocessedMsg, PersonalSignature.TYPE_OF_SIGNATURE);
    }

    public fromSignature( rawSignature: string) {
        super.signatureAndType(rawSignature, PersonalSignature.TYPE_OF_SIGNATURE);
    }

    public processMessage( unprocessedMsg: Uint8Array): number[] {
        return SignatureUtility.convertToPersonalEthMessage(uint8toString(unprocessedMsg));
    }

}

File: src/main/java/io/alchemynft/attestation/PersonalSignature.java
package io.alchemynft.attestation;

import org.bouncycastle.crypto.AsymmetricCipherKeyPair;
import org.tokenscript.attestation.core.SignatureUtility;

public class PersonalSignature extends AbstractSignature {
  private static final String TYPE_OF_SIGNATURE = "personal";

  public PersonalSignature(AsymmetricCipherKeyPair keys, byte[] unprocessedMsg) {
    super(keys, unprocessedMsg, TYPE_OF_SIGNATURE);
  }

  public PersonalSignature(byte[] rawSignature) {
    super(rawSignature, TYPE_OF_SIGNATURE);
  }

  @Override
  byte[] processMessage(byte[] unprocessedMsg) {
    return SignatureUtility.convertToPersonalEthMessage(unprocessedMsg);
  }

}
